src/main/scala/org/broadinstitute/dsde/vault/services/analysis/UpdateServiceHandler.scala

```diff
@@ -56,7 +65,12 @@ case class UpdateServiceHandler(requestContext: RequestContext, dmService: Actor
 
     case BossObjectCreated(bossObject: BossCreationObject, creationKey: String) =>
       bossObjects += creationKey -> bossObject.objectId.get
-      bossService ! BossClientService.BossResolveObject(BossDefaults.getResolutionRequest("PUT"), bossObject.objectId.get, creationKey)
+      // If the client is using the "X-Force-Location" header, no need to populate PUT urls.
+      // Instead, forward to the DM service if all BOSS objects have been created.
+      if (!forceLocationHeader)
+        bossService ! BossClientService.BossResolveObject(BossDefaults.getResolutionRequest("PUT"), bossObject.objectId.get, creationKey)
+      if (fileCount == bossObjects.size)
```

Comment:
We need an `else` here so `DMUpdateAnalysis` is not sent with `providedFiles` when `forceLocationHeader` is false.

Alternatively, we could nest another `if` in here to send `bossObjects` in the official-API case at this time, and rearrange the logic below to not send `DMUpdateAnalysis`.